hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```diff
@@ -476,17 +427,95 @@ public Response getDeletedKeySummary() {
   @GET
   @Path("/deletePending")
   public Response getDeletedKeyInfo(
-      @DefaultValue(DEFAULT_FETCH_COUNT) @QueryParam(RECON_QUERY_LIMIT)
-      int limit,
-      @DefaultValue(StringUtils.EMPTY) @QueryParam(RECON_QUERY_PREVKEY)
-      String prevKey) {
-    KeyInsightInfoResponse
-        deletedKeyInsightInfo = new KeyInsightInfoResponse();
-    getPendingForDeletionKeyInfo(limit, prevKey,
-        deletedKeyInsightInfo);
+      @DefaultValue(DEFAULT_FETCH_COUNT) @QueryParam(RECON_QUERY_LIMIT) int limit,
+      @DefaultValue(StringUtils.EMPTY) @QueryParam(RECON_QUERY_PREVKEY) String prevKey,
+      @DefaultValue(StringUtils.EMPTY) @QueryParam(RECON_QUERY_START_PREFIX) String startPrefix) {
+
+    // Initialize the response object to hold the key information
+    KeyInsightInfoResponse deletedKeyInsightInfo = new KeyInsightInfoResponse();
+
+    // Ensure the limit is non-negative
+    limit = Math.max(0, limit);
+
+    boolean keysFound = false;
+
+    try {
+      if (isNotBlank(startPrefix)) {
+        // Validate and apply prefix-based search
+        if (!validateStartPrefix(startPrefix)) {
+          return createBadRequestResponse("Invalid startPrefix: Path must be at the bucket level or deeper.");
+        }
+        keysFound = getPendingForDeletionKeyInfo(limit, prevKey, startPrefix, deletedKeyInsightInfo);
+      } else {
```

Comment:
We can do some optimization here. We may not need else here and just pass startPrefix as it is after validation if startPrefix is valid or not. Because whether startPrefix is empty or not, those cases are being handled in `extractKeysFromTable` method already.

Reply:
I agree, have made the changes.